Build fails on jdk 11

1.5 is specified as target release for compiler, while jdk 11 supports only 6 and above. 
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] error: Source option 5 is no longer supported. Use 6 or later.
[ERROR] error: Target option 1.5 is no longer supported. Use 1.6 or later.
```

--- a/pom.xml
+++ b/pom.xml
@@ -17,7 +17,7 @@
 
   <properties>
     <scala.major.version>2.11</scala.major.version>
-    <scala.minor.version>7</scala.minor.version>
+    <scala.minor.version>12</scala.minor.version>
     <scala.version>${scala.major.version}.${scala.minor.version}</scala.version>
     <scalatest.version>3.0.1</scalatest.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
@@ -30,8 +30,8 @@
     <maven.core.version>3.3.9</maven.core.version>
     <junit.version>4.12</junit.version>
     <maven.reporting.api.version>3.0</maven.reporting.api.version>
-    <scala.maven.plugin.version>3.2.2</scala.maven.plugin.version>
     <maven.enforcer.plugin.version>1.4.1</maven.enforcer.plugin.version>
+    <java.version>6</java.version>
   </properties>
 
   <licenses>
@@ -135,7 +135,7 @@
       <plugin>
         <groupId>net.alchim31.maven</groupId>
         <artifactId>scala-maven-plugin</artifactId>
-        <version>${scala.maven.plugin.version}</version>
+        <version>4.3.0</version>
         <configuration>
           <recompileMode>incremental</recompileMode>
           <scalaVersion>${scala.version}</scalaVersion>
@@ -152,10 +152,10 @@
       </plugin>
       <plugin>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>2.3.2</version>
+        <version>3.8.1</version>
         <configuration>
-          <source>1.5</source>
-          <target>1.5</target>
+          <source>1.${java.version}</source>
+          <target>1.${java.version}</target>
         </configuration>
       </plugin>
       <plugin>
@@ -270,7 +270,7 @@
           <docencoding>UTF-8</docencoding>
           <docfilessubdirs>true</docfilessubdirs>
           <show>package</show>
-          <source>1.5</source>
+          <source>1.${java.version}</source>
           <detectLinks>true</detectLinks>
         </configuration>
       </plugin>

